3scale-operator/tasks/main.yml

```diff
@@ -69,10 +71,28 @@
     - name: Adding result to list
       ansible.builtin.set_fact:
         test_results: "{{ test_results + [false] }}"
-
-- name: Create APICast
+- name: Create Prerequisites for APICast Operator
   block:
-    - name: Create APICast in target_namespace
+    - name: Create Subscription to install APICast Operator
+      kubernetes.core.k8s:
+        state: present
+        namespace: "{{ target_namespace }}"
+        definition:
+          apiVersion: operators.coreos.com/v1alpha1
+          kind: Subscription
+          metadata:
+            labels:
+              operators.coreos.com/apicast-operator.3scale: ""
+            name: apicast-operator
+          spec:
+            channel: threescale-2.13
+            installPlanApproval: Automatic
+            name: apicast-operator
+            source: redhat-operators
+            sourceNamespace: openshift-marketplace
+            startingCSV: apicast-operator.v0.7.4
+      failed_when: result.resources[0] is not defined
```

Comment:
this will not work here because although we POST the subscription object and see the resulting CSV (and successful operator install), we can't actually GET subscription objects on ROSA. Not sure if it's a permissions thing with the cluster-admin account we use for running these reports.

You should fail if we don't see the ApiCast CSV in Phase Succeeded.

Reply:
You do get the result back. But here, since you need to wait for the subscription to finish, there will need to be a loop to check for that. Subscription can't use wait_condition.